### PR TITLE
[Ready to merge] Added unit test for oddness inference in ternary integer products

### PR DESCRIPTION
**Pull request by toolforger**
Tests issue #9127 .

### CODE REVIEW

**Comment by toolforger** [comment]
Turned out another test would hit the exact same issue.

**Comment by toolforger** [comment]
... aaand for the old assumptions as well.
Well, that could have gone to another PR, but seeing this one hasn't merged yet, I'm just including the changes here.

**Comment by toolforger** [comment]
Aww, my bad - I added a commit for a separate (but similar) issue.
Forgot to start a new branch :-(
Hopefully this can go in anyway.

**Comment by debugger22** [comment]
Perhaps, you shouldn't duplicate test names.(Different for both assumptions)

**Comment by toolforger** [comment]
Can you name an example? I'm not sure which of several potentially odd-looking things you mean, I hope an example will put me on track with that.

**Comment by debugger22** [comment]
`./bin/test -k test_evenness_in_ternary_integer_product_with_odd` resulted in execution of two tests because they are defined twice in different files.

**Comment by debugger22** [comment]
Oops wrong button!

File: sympy/core/tests/test_arit.py [hunk 1]
```diff
@@ -435,14 +435,48 @@ def test_Mul_is_even_odd():
...
+@XFAIL
+def test_oddness_in_ternary_integer_product_with_odd():
+    # Tests that oddness inference is independent of term ordering.
+    # Term ordering at the point of testing depends on SymPy's symbol order, so
+    # we try to force a different order by modifying symbol names.
+    x = Symbol('x', integer=True)
+    y = Symbol('y', integer=True)
+    k = Symbol('k', odd=True)
     assert (x*y*(y + k)).is_odd is False
+    assert (y*x*(x + k)).is_odd is False
```
Comment (debugger22) [inline sympy/core/tests/test_arit.py]: Maybe a little comment here about which one works and which doesn't would help.

Comment (toolforger) [inline sympy/core/tests/test_arit.py]: One of them will always fail, and one of them will always succeed, depending on internal term ordering.
It's exposing the bug described in #9127 (that's why the test is on `@XFAIL`).
Should I mention the issue number on the `@XFAIL` line?

**Comment by toolforger** [comment]
Am 13.04.2015 um 22:16 schrieb Sudhanshu Mishra:
> `./bin/test -k test_evenness_in_ternary_integer_product_with_odd` resulted in execution of two tests because they are defined twice in different files.

That's intentional - both tests refer to the same problem, one 
implemented in old assumptions and one implemented in new assumptions 
(it's a case of a logic error that was consistently applied to both 
systems).
So if you want to test this particular issue, do `bin/test -k ...`. With 
`bin/test arit` you test just one of the assumption systems, with 
`bin/test arith -k ...` you run just one test in one of the assumption 
systems.

File: sympy/core/tests/test_arit.py (hunk 1 above)
Comment (debugger22) [inline sympy/core/tests/test_arit.py]: No, I mean something like this:
```
assert (x*y*(y + k)).is_even is True  # This works
assert (y*x*(x + k)).is_even is True  # This doesn't
```
It's not necessary, though.

Comment (toolforger) [inline sympy/core/tests/test_arit.py]: Until #9127 is fixed, one of them will always work, the other will always fail.
Which of them will work and which will fail depends on term sort order, which will change once I finish another project that I'm working on. So it's technically impossible to add that kind of comment.

Comment (debugger22) [inline sympy/core/tests/test_arit.py]: Oh! Understood, thanks!

**Pull request merged**